addons/udes_stock/models/stock_picking.py

```diff
@@ -2753,3 +2753,46 @@ def _get_pick_details(self):
             "submitted_value": self.origin,
         }
         return pick_details
+
+    @api.model
+    def _set_original_picking_on_historical_pickings(self):
+        """
+        Set the Original Picking for historical pickings that do not have an Original Picking set.
+
+        Carries out update as superuser to avoid any permissions issues, as the method is simply
+        setting Original Picking to the correct value, and doesn't take any input variables.
+        """
+        Picking = self.env["stock.picking"]
+
+        pickings_to_update = Picking.sudo().with_context(prefetch_fields=False).search(
+            [("u_original_picking_id", "=", False)], order="id"
+        )
+
+        picking_ids = pickings_to_update.ids
+        picking_count = len(picking_ids)
+
+        if picking_count:
+            _logger.info("Setting Original Picking value on %s pickings", picking_count)
+
+            update_datetime = fields.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S.%f")
+
+            update_query = """
+            UPDATE stock_picking
+            SET
+            write_uid = {user_id},
+            write_date = '{update_datetime}',
```

Comment:
`write_date = now() at time zone 'UTC'` would be consistent with how [Odoo updates write date](https://github.com/unipartdigital/odoo/blob/ee08a6f63711c950bc6c3eae61fd7f959824c0c4/odoo/models.py#L3504) (unless we specifically want the application time, instead of commit time?).

Reply:
Originally I did it like this to make sure all pickings updated had the same datetime for the write date, but at that point pickings were being updated in batches.

With them all now being updated in one query I should be able to use this, will let you know how I get on. :)

Reply:
`now() at ...` and similar will always be the same for all queries within a single transaction - see the send of [this section](https://www.postgresql.org/docs/14/functions-datetime.html#FUNCTIONS-DATETIME-CURRENT)

Reply:
Updated now, good spot. :)